Automattic/wp-calypso
# Jetpack Plugins Management: Use a bulk features endpoint for loading the features of all sites

#### Proposed Changes

The agency dashboard needs to scale for agencies with many hundreds of sites. Currently it has performance issues for agencies with >500 sites that make it unusable. This PR is the first pass on performance improvements targeting these large agencies. It does not fix all performance issues but it does improve it significantly.

Changes included:
1. Features are now fetched from a bulk endpoint instead of calling an endpoint once per site.
2. Redux state is updated once for both plugins and features instead of being updated once per site.
3. The `getPlugins` selector is memoized.


#### Testing Instructions



1. Visit `/plugins/manage` as an agency with many sites. See this Slack comment for an example: p1669769472607889-slack-C039K7CD119
2. Verify that performance is improved and nothing is broken.

#### Pre-merge Checklist





Related to 1202619025189113-as-1203463932378052
